src/renderer/containers/query-browser.jsx

```diff
@@ -154,6 +157,31 @@ class QueryBrowserContainer extends Component {
     dispatch(refreshDatabase(database));
   }
 
+  onShowDiagramModal(database) {
+    const { dispatch } = this.props;
+    dispatch(showDatabaseDiagram(database.name));
+  }
+
+  onShowDatabaseDiagram(database) {
+    const { dispatch } = this.props;
+    const selectedTables = [];
+
+    $(':checkbox:checked', 'div.ui.list').map((index, checkbox) => {
+      selectedTables.push(checkbox.id);
+    });
```

Comment:
This is actually not the same. Your code would contain objects that jQuery returns. I'm using jQuery here because of easier reading checked checkboxes. I guess reading refs of the refs of the refs (react way) would be more complicated for only reading which checkbox is selected? Though I agree jQuery way might be replaced for react way of handling checkboxes in `database-diagram-modal.jsx`